File: C_string/rs/src/c_string/c7.rs
pub struct ReplaceReturn {
    pub result: String,
    pub replacements: u32,
    pub original: String,
}

pub fn replace_text(text: &str, old_text: &str, new_text: &str) -> ReplaceReturn {
    let replacements = text.matches(old_text).count() as u32;
    let result = text.replace(old_text, new_text);

    ReplaceReturn {
        result,
        replacements,
        original: text.to_string(),
    }
}

File: challenge/C_string/js/c7.js
function replaceText(text, oldText, newText) {
    const regex = new RegExp(oldText, "g");
    const matches = text.match(regex);
    const replacements = matches ? matches.length : 0;
    const result = text.replace(regex, newText);
    
    return { result, replacements, original: text };
}

// Driver code
console.log(replaceText("Hello world hello", "hello", "hi"));
console.log(replaceText("test test test", "test", "exam"));
console.log(replaceText("JavaScript", "Python", "Java"));

